test/test_torch.py

```diff
@@ -4303,14 +4303,72 @@ def test_func(slf, device, call_type):
         test_func(self, device, 'method inplace')
 
     def test_index_fill(self, device):
+        def _verify_against_expected(x, idx, val, expected):
```

Comment:
How hard would it be to implement `torch.Tensor.index_fill` using NumPy operators? That would let us add a reference to the `index_fill` OpInfo and this test could just validate against the NumPy implementation, which would be simpler and more readable.